src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala

```diff
@@ -65,7 +65,14 @@ trait ShellConfig {
   // This property is used in TypeDebugging. Let's recycle it.
   val colorOk = Properties.coloredOutputEnabled
 
-  val historyFile = s"$userHome/.scala_history_jline3"
+  val datadir =
+    try {
+      val projectdirs = ProjectDirectories.from("org", "scala-lang", "repl2")
+      projectdirs.dataLocalDir
+    } catch {
+      case _: UnsupportedOperationException => userHome
+    }
+  val historyFile = s"$datadir/.scala_history"
```

Comment:
I'm dubious about omitting the `_jline3` here. Maybe it's okay, but I would need convincing.

We added it when we moved from JLine 2 to JLine 3 because the format is different. So there is a risk of JLine 2 failing to understand the JLine 3 format, and vice versa. Perhaps even a risk of corruption.

Do we believe that it can be omitted because the file will always be in a different directory than any history file associated with a Scala version that uses JLine 2...? Even if so, I'm not sure it's sufficient reassurance.

Finally, Scala 2 generally is in a "don't fix it if ain't broke" kind of state, and since we've been using `.scala_history_jline3` for a while now, I'm inclined to leave it alone unless convinced otherwise.

Reply:
I agree, I jumped the gun on utopian naming.